lib/simulator.js

```diff
@@ -985,44 +985,75 @@ function launch(simHandleOrUDID, options, callback) {
 									return next(new Error(__('Simulator is not available')));
 								}
 
-								if (/^shutdown/i.test(sim.state)) {
-									// the udid that is supposed to be running isn't, kill the simulator
-									emitter.emit('log-debug', __('%s Simulator is running, but udid %s is shut down, stopping simulator', handle.name, handle.udid));
-									stop(handle, next);
-									return;
-								}
+								function waitToBoot() {
+									emitter.emit('log-debug', __('Waiting for sim to boot...'));
+									simctl.waitUntilBooted({ simctl: handle.simctl, udid: handle.udid, timeout: 30000 }, function (err, _booted) {
```

Comment:
Might this be a good time to launch the Simulator earlier like some (..) other platforms do? Basically trigger it as soon as we get the UUID, so we won't have any delay for the initial launch anymore.

Reply:
We can't launch the simulator and then later install the app. ioslib's API isn't designed to do this. I can fix it, but I'd rather allocate the time towards ioslib v2 where I'll fix all these issues.